### ISSUE TITLE
Feature Idea: Resume workflow from point of failure

### ISSUE THREAD

**Issue by MarkRoddick** [ansible/awx#1284]
##### ISSUE TYPE
 - Feature Idea

##### COMPONENT NAME
 - API
 - UI

##### SUMMARY
Currently you can only rerun individual templates that have executed or failed as part of the workflow. It would be nice to be able to restart from failure and continue with the rest of the workflow.

**Comment by AlanCoding** [comment]
A workflow's execution can take multiple branches simultaneously, and some nodes could have already traversed their "failure" or "always" connections. The best feature spec I could imagine would be to start again on all filed nodes that ended their branch. There's appeal as an analog to retry on failed hosts for job templates, but I think it would be difficult to communicate its function.

**Comment by ghost** [comment]
Any update on this feature?

**Comment by cvernooy23** [comment]
Another way to do it is to leave it up to users. Select the template to start at and run from there with the understanding that any variables that were set in previous jobs would not exist anymore and that some branches could have already been passed. This way operators of the system can decide which is the better option for the specific workflow, start from the beginning or continue.

**Comment by sumkincpp** [comment]
+1 On this feature. Definitely must have in roadmap.

After solving all necessary problems engineer goes to specific failure node job(linked with wf) and relaunches it, thus continuing workflow execution.

**Comment by boagg** [comment]
+1

**Comment by spsingh1982** [comment]
When is this getting released?

**Comment by ryanpetrello** [comment]
@spsingh1982 whenever somebody writes the code 😉

**Comment by ravitejajyosula** [comment]
+1

**Comment by bobmacks** [comment]
+1 - So important to be resume from point of failure for large workflows with many nodes

**Comment by Slm0n87** [comment]
+1

**Comment by sub6as** [comment]
damn, I was hoping this was a thing already, nevermind.

+1

**Comment by Joey-ibm** [comment]
+1

**Comment by grcunha** [comment]
+1

**Comment by kvprasoon** [comment]
I think giving a 👍 on the Idea will be better than a comment with +1. We can avoid mail storm :-)

**Comment by shanemcd** [comment]
PSA: for each "+1" comment, we'll ignore the request for 1 more year. Looks like the ETA for this one is 2026.

**Comment by Gratizzz** [comment]
Is it on the roadmap? Really need this feature as well.

**Comment by hgonzalerh** [comment]
I'm seeing interest in this from large organizations with multi-hour workflows. +1

**Comment by picchetts** [comment]
+1

**Comment by 2and3makes23** [comment]
Would it be possible to allow graph cycles with approval nodes to achieve something similar to "resume after failure"?

This would enable placing an approval node behind critical actions that trigger on failure and lead back to the failed node like this:
![Image](https://github.com/user-attachments/assets/f153667d-27de-4ec1-99a5-ceefe00e7152)
If an error occurs the workflow would wait for approval to retry from the formerly failed node.

**Comment by httran13** [comment]
+1

**Comment by httran13** [comment]
> Would it be possible to allow graph cycles with approval nodes to achieve something similar to "resume after failure"?
> 
> This would enable placing an approval node behind critical actions that trigger on failure and lead back to the failed node like this: ![Image](https://github.com/user-attachments/assets/f153667d-27de-4ec1-99a5-ceefe00e7152) If an error occurs the workflow would wait for approval to retry from the formerly failed node.

i tried this, maybe i have it wrong but its giving me a cycle detected error
  localhost failed | msg: Failed to associate item {'Error': 'Cycle detected.'}


Node A 
Node B
Approval Node C
Node D

Node B success > Node D
Node B failure > Approval Node C

Approval Node C success > Node B

Is that correct? im using ansible awx.awx collection

**Comment by 2and3makes23** [comment]
> i tried this, maybe i have it wrong but its giving me a cycle detected error localhost failed | msg: Failed to associate item {'Error': 'Cycle detected.'}

Yes, cycles are currently not supported (intentionally). Above I wanted to question if the developers would want to consider allowing cycles with approval nodes (that are set to require manual intervention) in order to gain the ability to restart failed workflows from certain points.
